network/src/receiver.rs

```diff
@@ -69,11 +71,18 @@ impl<Handler: MessageHandler> Receiver<Handler> {
     async fn spawn_runner(socket: TcpStream, peer: SocketAddr, handler: Handler) {
         tokio::spawn(async move {
             let transport = Framed::new(socket, LengthDelimitedCodec::new());
-            let (mut writer, mut reader) = transport.split();
+            let (writer, mut reader) = transport.split();
+
+            let (tx_response, tr_response): (Writer, mspcReceiver<Bytes>) = channel(100);
+            Self::spawn_response_writer(tr_response, writer).await;
+
             while let Some(frame) = reader.next().await {
                 match frame.map_err(|e| NetworkError::FailedToReceiveMessage(peer, e)) {
                     Ok(message) => {
-                        if let Err(e) = handler.dispatch(&mut writer, message.freeze()).await {
+                        if let Err(e) = handler
```

Comment:
There is an alternative to channels if we are OK with the handler returning a single value (look to my comment above mentioning a `oneshot::channel`). 

```rust
pub struct Outbound<F> {
    future: F,
    writer: TcpWriter,
}

impl<F> Outbound<F> {
    pub fn new(future: F, writer: TcpWriter) -> Self {
        Self { future, writer }
    }
}

impl<F> Future for Outbound<F>
where F: Future<Output = io::Result<Bytes>> + Unpin
{
    type Output = io::Result<())>;

    fn poll(mut self: Pin<&mut Self>, cx: &mut Context<'_>) -> Poll<Self::Output> {
        let response = ready!(Pin::new(&mut self.future).poll(cx))?;
        self.writer.send(response).await
        Poll::Ready(Ok(()))
    }
}
```

Reply:
@huitseeker Are there any other steps a contributor should go through, such as running the benchmarks, when working on Narwhal?

Reply:
> Are there any other steps a contributor should go through, such as running the benchmarks, when working on Narwhal?

That's it. Of course, we should have integration tests (#75), but the (local) benchmark serves that purpose for us until we do.